packages/iTwinUI-react/src/core/Table/hooks/useResizeColumns.tsx

```diff
@@ -280,12 +276,20 @@ const reducer = <T extends Record<string, unknown>>(
     );
 
     if (
-      !isNewColumnWidthsValid(newColumnWidths, instance?.flatHeaders) ||
-      !isNewColumnWidthsValid(newNextColumnWidths, instance?.flatHeaders)
+      !isNewColumnWidthsValid(newColumnWidths, instance) ||
+      !isNewColumnWidthsValid(newNextColumnWidths, instance)
     ) {
       return newState;
     }
 
+    // Setting `width` here because it might take several rerenders until actual column width is set.
+    // Also setting after the actual resize happened.
+    instance?.flatHeaders.forEach((h) => {
```

Comment:
Moved here from pointer down handler because it was causing a bug when nothing was resized because e.g. you try to reduce size of a column that already is in its minimum width.
![table-bug-resize](https://user-images.githubusercontent.com/36186912/191452212-f3c35c1e-6ed5-410f-92b0-a14e607e4393.gif)